lib/mvi/responses/base.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+module MVI
+  module Responses
+    class Base
+      attr_accessor :code, :query, :original_response
+
+      RESPONSE_CODES = {
+        success: 'AA',
+        failure: 'AE',
+        invalid_request: 'AR'
+      }.freeze
+
+      class << self
+        attr_accessor :endpoint
+      end
+
+      def self.mvi_endpoint(endpoint)
```

Comment:
@omgitsbillryan the idea was to make subclass config DSL like, @austin-martinez I agree it does make the base class more complex, we have a similar pattern for the redis store.